cpp/examples/ide_secir_ageres.cpp

```diff
@@ -0,0 +1,136 @@
+#include "memilio/data/analyze_result.h"
+
+int main()
+{
+    using Vec = mio::TimeSeries<ScalarType>::Vector;
+
+    int num_agegroups = 2;
+
+    ScalarType tmax                = 5;
+    std::vector<ScalarType> N      = std::vector<ScalarType>(num_agegroups, 5000.);
+    std::vector<ScalarType> deaths = std::vector<ScalarType>(num_agegroups, 6.);
+    ScalarType dt                  = 1.;
+
+    int num_transitions = (int)mio::isecir::InfectionTransition::Count;
+
+    // Create TimeSeries with num_transitions * num_agegroup elements where transitions needed for simulation will be stored.
+    mio::TimeSeries<ScalarType> init(num_transitions * num_agegroups);
+
+    // Add time points for initialization of transitions.
+    Vec vec_init(num_transitions * num_agegroups);
+    // First AgeGroup.
+    vec_init[(int)mio::isecir::InfectionTransition::SusceptibleToExposed]                 = 25.0;
+    vec_init[(int)mio::isecir::InfectionTransition::ExposedToInfectedNoSymptoms]          = 15.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms] = 8.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedNoSymptomsToRecovered]        = 4.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSymptomsToInfectedSevere]     = 1.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSymptomsToRecovered]          = 4.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSevereToInfectedCritical]     = 1.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedSevereToRecovered]            = 1.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedCriticalToDead]               = 1.0;
+    vec_init[(int)mio::isecir::InfectionTransition::InfectedCriticalToRecovered]          = 1.0;
+    //Second AgeGroup.
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::SusceptibleToExposed]                 = 25.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::ExposedToInfectedNoSymptoms]          = 15.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedNoSymptomsToInfectedSymptoms] = 8.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedNoSymptomsToRecovered]        = 4.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedSymptomsToInfectedSevere]     = 1.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedSymptomsToRecovered]          = 4.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedSevereToInfectedCritical]     = 1.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedSevereToRecovered]            = 1.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedCriticalToDead]               = 1.0;
+    vec_init[num_transitions + (int)mio::isecir::InfectionTransition::InfectedCriticalToRecovered]          = 1.0;
```

Comment:
Here you are using the same values for both age groups. Can you create one vector for one age group and then set vec_init by using this vector for both age groups? using .begin() and .end() could be an option for this. I think this would make the example easier to understand.